### PR TITLE
chore(ci) : cache go-build folders

### PR DESCRIPTION
**Pull request by mmorel-35**
Signed-off-by: Matthieu MOREL <mmorel-35@users.noreply.github.com>

### CODE REVIEW

**Comment by willnorris** [comment]
Here's an interesting cross platform approach that uses the output of `got env GOCACHE` and `go env GOMODCACHE`: https://markphelps.me/2019/11/speed-up-your-go-builds-with-actions-cache/

what do you think?  he has it split into two separate cache steps, but I'm not sure that' necessary, since they're both using the hash of `**/go.sum` for the cache key.

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -43,12 +38,16 @@ jobs:
       with:
         go-version: ${{ matrix.go-version }}
     - uses: actions/checkout@v2
-
+    
+    # Get values for cache paths to be used in later steps 
+    - id: go-cache
+      run: echo "::set-output name=go-build::$(go env GOCACHE)"
```
Comment (willnorris) [inline .github/workflows/tests.yml:44]: might as well use this approach for `go env GOMODCACHE` too?  I mean, it's not likely to change, but you never know and the consistency is nice.

Comment (mmorel-35) [reply .github/workflows/tests.yml]: I agree

**Comment by willnorris** [comment]
sweet, thanks!  I'll let these latest round of tests finish running then merge.

**Pull request merged**